src/Input.h

```diff
@@ -115,6 +153,8 @@ class Input {
 	std::map<std::string, BindingPage> bindingPages;
 	std::map<std::string, KeyBindings::ActionBinding> actionBindings;
 	std::map<std::string, KeyBindings::AxisBinding> axisBindings;
+
+	std::list<InputFrame *> inputFrames;
```

Comment:
I don't see any `new` or `delete` around, so there's a reason preferring that way
instead of:
`std::list<InputFrame &> inputFrames;`

(Note compiler will allow `delete (InputFrame *) ptr_to_InputFrame;` ...)

Second question is: why a `std::list` and not a `std::vector`?

Reply:
InputFrames are managed by other code not owned by the Input system. It is expected (and required) that the code responsible for pushing an InputFrame pops the frame before destroying it. However, the non-nullable property of a reference would be better. I'll get that fixed up. (I'm much more fluent in C/Lua than C++ - I forgot references were a storable thing.)

`std::list` was chosen for the constant-time insertion / deletion property, but considering that it's something small like a reference, I will probably swap it to a `std::vector`.

Thanks for the review!

Reply:
EDIT: I also didn't know that STL containers hate references and they're more trouble than they're worth. Pointers it is!

Reply:
> EDIT: I also didn't know that STL containers hate references and they're more trouble than they're worth. Pointers it is!

Sorry, I had to check my brain was on :/ ...